src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -416,7 +428,54 @@ private void RenderUnboundAttribute(string attributeName, Chunk attributeValueCh
                 // C# code, then we need to buffer it.
                 if (!isPlainTextValue)
                 {
-                    BuildBufferedWritingScope(attributeValueChunk, htmlEncodeValues: true);
+                    BuildBufferedWritingScope(renderValue: () =>
+                    {
+                        IChunkVisitor visitor;
+                        var parentChunk = attributeValueChunk as ParentChunk;
+                        containsDynamicCodeAttributeChunk = parentChunk
+                            .Children
```

Comment:
What's the point of the conditional cast if you're going to dereference the result immediately?